tests/test_dpd_builder.py

```diff
@@ -53,3 +54,90 @@ def test_all_subsystems(
+                    R"\mathcal{H}^\mathrm{\overline{\Sigma}(1660)^{-}}[1/2, -1/2, -1/2, 0]",
+                    R"\mathcal{H}^\mathrm{\overline{\Sigma}(1660)^{-}}[1/2, -1/2, 1/2, 0]",
+                    R"\mathcal{H}^\mathrm{\overline{\Sigma}(1660)^{-}}[1/2, 1/2, -1/2, 0]",
+                    R"\mathcal{H}^\mathrm{\overline{\Sigma}(1660)^{-}}[1/2, 1/2, 1/2, 0]",
+                ]
+            else:  # CANONICAL BASE
+                assert n_coupling_symbols == 4
+                assert coupling_symbols_str == [
+                    R"\mathcal{H}^\mathrm{LS,N(1700)^{+}}[1, 1, 2, 1/2]",
+                    R"\mathcal{H}^\mathrm{LS,N(1700)^{+}}[1, 2, 2, 1/2]",
+                    R"\mathcal{H}^\mathrm{LS,\overline{\Sigma}(1660)^{-}}[0, 1, 1, 1/2]",
+                    R"\mathcal{H}^\mathrm{LS,\overline{\Sigma}(1660)^{-}}[2, 1, 1, 1/2]",
+                ]
+        else:  # noqa: PLR5501
+            if min_ls:  # HELICITY BASE
+                assert n_coupling_symbols == 14
+                assert coupling_symbols_str == [
+                    R"\mathcal{H}^\mathrm{decay}[N(1700)^{+}, 0, -1/2]",
+                    R"\mathcal{H}^\mathrm{decay}[N(1700)^{+}, 0, 1/2]",
+                    R"\mathcal{H}^\mathrm{decay}[\overline{\Sigma}(1660)^{-}, -1/2, 0]",
+                    R"\mathcal{H}^\mathrm{decay}[\overline{\Sigma}(1660)^{-}, 1/2, 0]",
+                    R"\mathcal{H}^\mathrm{production}[N(1700)^{+}, -1/2, -1/2]",
+                    R"\mathcal{H}^\mathrm{production}[N(1700)^{+}, -1/2, 1/2]",
+                    R"\mathcal{H}^\mathrm{production}[N(1700)^{+}, -3/2, -1/2]",
+                    R"\mathcal{H}^\mathrm{production}[N(1700)^{+}, 1/2, -1/2]",
+                    R"\mathcal{H}^\mathrm{production}[N(1700)^{+}, 1/2, 1/2]",
+                    R"\mathcal{H}^\mathrm{production}[N(1700)^{+}, 3/2, 1/2]",
+                    R"\mathcal{H}^\mathrm{production}[\overline{\Sigma}(1660)^{-}, -1/2, -1/2]",
+                    R"\mathcal{H}^\mathrm{production}[\overline{\Sigma}(1660)^{-}, -1/2, 1/2]",
+                    R"\mathcal{H}^\mathrm{production}[\overline{\Sigma}(1660)^{-}, 1/2, -1/2]",
+                    R"\mathcal{H}^\mathrm{production}[\overline{\Sigma}(1660)^{-}, 1/2, 1/2]",
+                ]
+            else:  # CANONICAL BASE
+                assert n_coupling_symbols == 6
+                assert coupling_symbols_str == [
+                    R"\mathcal{H}^\mathrm{LS,decay}[N(1700)^{+}, 2, 1/2]",
+                    R"\mathcal{H}^\mathrm{LS,decay}[\overline{\Sigma}(1660)^{-}, 1, 1/2]",
+                    R"\mathcal{H}^\mathrm{LS,production}[N(1700)^{+}, 1, 1]",
+                    R"\mathcal{H}^\mathrm{LS,production}[N(1700)^{+}, 1, 2]",
+                    R"\mathcal{H}^\mathrm{LS,production}[\overline{\Sigma}(1660)^{-}, 0, 1]",
+                    R"\mathcal{H}^\mathrm{LS,production}[\overline{\Sigma}(1660)^{-}, 2, 1]",
+                ]
```

Comment:
@Zeyna777 there is something I don't understand about these test cases.
- Why is the number of LS couplings smaller than the number of helicity couplings?
- Why is the number of coefficients (i.e. products of couplings) smaller than the number of couplings in the case of the canonical basis and larger in the helicity basis?